### PR TITLE
Upgrade analyzers to .NET Standard 2.0 and test them on .NET Core

### PR DESCRIPTION
**Pull request by thomaslevesque**
Cherry-picked the commits from my current work on https://github.com/FakeItEasy/FakeItEasy/issues/1662, I hope it didn't depend on anything I didn't include...

### CODE REVIEW

File: src/FakeItEasy.Analyzer.Shared/ArgumentConstraintAnalyzerBase.cs [hunk 1]
```diff
@@ -21,6 +21,8 @@ public override void Initialize(AnalysisContext context)
                 throw new ArgumentNullException(nameof(context));
             }
 
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.Analyze | GeneratedCodeAnalysisFlags.ReportDiagnostics);
+            context.EnableConcurrentExecution();
```
Comment (thomaslevesque) [inline src/FakeItEasy.Analyzer.Shared/ArgumentConstraintAnalyzerBase.cs]: Suggested by the "meta-analyzer"!

File: src/FakeItEasy.Analyzer.Shared/ArgumentConstraintTypeMismatchCodeFixProvider.cs [hunk 2]
```diff
@@ -51,27 +51,32 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
...
-                        ct => ChangeConstraintTypeAsync(context, diagnostic, ct)),
+                        ct => ChangeConstraintTypeAsync(context, diagnostic, ct),
+                        equivalenceKey: "ChangeConstraintType"),
                     diagnostic);
             }
 
             return CompletedTask;
         }
 
+        public override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
```
Comment (thomaslevesque) [inline src/FakeItEasy.Analyzer.Shared/ArgumentConstraintTypeMismatchCodeFixProvider.cs]: Do I need to say it again?

File: src/FakeItEasy.Analyzer.Shared/ArgumentConstraintAnalyzerBase.cs (hunk 1 above)
Comment (blairconrad) [reply src/FakeItEasy.Analyzer.Shared/ArgumentConstraintAnalyzerBase.cs]: What is the meta-analyzer?

Comment (thomaslevesque) [reply src/FakeItEasy.Analyzer.Shared/ArgumentConstraintAnalyzerBase.cs]: Sorry, I mean the "analyzer for writing analyzers". https://github.com/dotnet/roslyn-analyzers#microsoftcodeanalysisanalyzers

**Comment by thomaslevesque** [comment]
This change has been released as part of [FakeItEasy 5.4.0](https://github.com/FakeItEasy/FakeItEasy/releases/tag/5.4.0).

**Pull request merged**